lucille-core/src/main/java/com/kmwllc/lucille/indexer/SolrDocRequests.java

```diff
@@ -0,0 +1,62 @@
+package com.kmwllc.lucille.indexer;
+
+import com.kmwllc.lucille.core.Document;
+import org.apache.solr.common.SolrInputDocument;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class SolrDocRequests {
+  private final List<SolrInputDocument> solrAddUpdateDocs;
+  private final Set<String> solrAddUpdateIds;
+  private final Set<String> solrDeleteIds;
+
+  public SolrDocRequests() {
+    this(new ArrayList<>(), new HashSet<>(), new HashSet<>());
+  }
+
+  public SolrDocRequests(
+      List<SolrInputDocument> solrAddUpdateDocs,
+      Set<String> solrAddUpdateIds,
+      Set<String> solrDeleteIds) {
+    this.solrAddUpdateDocs = solrAddUpdateDocs;
+    this.solrAddUpdateIds = solrAddUpdateIds;
+    this.solrDeleteIds = solrDeleteIds;
+  }
+
+  public void addDocForAddUpdate(SolrInputDocument doc) {
+    this.solrAddUpdateDocs.add(doc);
```

Comment:
"this." isn't needed here -- this is outside a constructor and there's no ambiguity about solrAddUpdateDocs (it can only be the instance variable with that name)... same comment for the other methods in this class